html/browsers/the-window-object/apis-for-creating-and-navigating-browsing-contexts-by-name/open-features-tokenization-001.html

```diff
@@ -0,0 +1,148 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>HTML: window.open `features`: tokenization -- `noopener`</title>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/#apis-for-creating-and-navigating-browsing-contexts-by-name">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+var windowURL = 'resources/close-self.html';
+
+// Tests for how windows features are tokenized into 'name', 'value'
+// window features separators are ASCII whitespace, '=' and  ','
+
+test (t => {
+  // Tokenizing `name`: initial window features separators are ignored
+  // Each of these variants should tokenize to ('noopener', '')
+  var featureVariants = [
+    ' noopener',
+    '=noopener',
+    ',,noopener',
+    ',=, noopener',
+    '\n=noopener=',
+    '\tnoopener',
+    '\r,,,=noopener',
+    '\u000Cnoopener'
+  ];
+  featureVariants.forEach(feature => {
+    var win = window.open(windowURL, '', feature);
+    assert_equals(win, null, `"${feature}" should activate feature "noopener"`);
+  });
+}, 'tokenization should skip window features separators before `name`');
+
+test (t => {
+  // Tokenizing `name`: lowercase conversion
+  // Each of these variants should tokenize as feature ('noopener', '')
+  // except where indicated
+  // Note also that `value` is lowercased during tokenization
+  var featureVariants = [
+    'NOOPENER',
+    'noOpenER',
+    '  NOopener',
+    '=NOOPENER',
+    'noopener=NOOPENER', // => ('noopener', 'noopener')
+    'NOOPENER=noopener' // => ('noopener', 'noopener')
+  ];
+  featureVariants.forEach(feature => {
+    var win = window.open(windowURL, '', feature);
+    assert_equals(win, null, `"${feature}" should activate feature "noopener"`);
+  });
+}, 'feature `name` should be converted to ASCII lowercase');
+
+test (t => {
+  // After `name` has been collected, ignore any window features separators until '='
+  // except ',' OR a non-window-features-separator — break in those cases
+  // i.e. ignore whitespace until '=' unless a ',' is encountered first
+  // Each of these variants should tokenize as feature ('noopener', '')
+  var featureVariants = [
+    'noopener',
+    ' noopener\r',
+    'noopener\n =',
+    'noopener,',
+    'noopener  =,',
+    ', noopener   =',
+    'noopener,=',
+    'noopener foo', // => ('noopener', ''), ('foo', '')
+    'foo noopener=1', // => ('foo', ''), ('noopener', '1')
+    'foo=\u000Cnoopener' // => ('foo', ''), ('noopener', '')
```

Comment:
I am unclear why this should activate noopener feature. I personally would have expected the result to be ('foo', 'noopener'). I would think the loop at 7.1 (https://html.spec.whatwg.org/#concept-window-open-features-tokenize) would skip past both the '=' and the '\u000C' since both are feature separators. Could someone please clarify?